### PR TITLE
Add LatentPoolCTRNN model for multi-dataset training (Issue #26)

### LINKED ISSUE #26: Based on current CTRNN model, add a new model with a large number of "latent" neurons
- Add functionality to train this pool of neurons on activity from multiple recording datasets with variable number of observed neurons
- This training functionality should also account for multiple "stimulus" neurons to indicate what stimulus was given
- Input neurons should take continuous input to indicate the concentration of the delivered stimulus
- The goal of the system is to estimate smooth firing rates given input binned spike counts

### PR DESCRIPTION
**Pull request by abuzarmahmood**
## Summary
This PR implements a new LatentPoolCTRNN model that addresses all requirements from issue #26.

## Changes
- **New Model**: LatentPoolCTRNN class in src/model.py
  - Large pool of latent neurons (configurable, default 256)
  - Flexible encoder/decoder architecture for variable neuron counts
  - Integrated stimulus type (one-hot encoded) and concentration (continuous) inputs
  - Built on top of existing CTRNN architecture

- **Training Function**: train_latent_pool_model in src/train.py
  - Supports training on multiple datasets simultaneously
  - Handles variable numbers of observed neurons via masking
  - Cross-validation and early stopping
  - Model checkpointing

- **Example Script**: src/run_latent_pool_model.py
  - Complete demonstration with synthetic data
  - Shows how to prepare data for multiple datasets
  - Includes evaluation metrics (R² scores)
  - Generates training plots

## Key Features
- Large pool of latent neurons that learn shared dynamics across datasets
- Handles multiple recording datasets with variable numbers of observed neurons
- Multiple stimulus neurons indicating stimulus type (one-hot encoded)
- Continuous input for stimulus concentration
- Estimates smooth firing rates from binned spike counts
- Masking mechanism to handle padding for variable neuron counts

## Architecture
Input: [spike_counts, stimulus_type, stimulus_concentration]
  -> Encoder (with dropout)
  -> Large Latent CTRNN (learns shared dynamics)
  -> Decoder (with dropout)
  -> Output: smooth firing rates

## Usage Example
See src/run_latent_pool_model.py for a complete working example with synthetic data.

## Testing
Run the example script to see the model in action:
python src/run_latent_pool_model.py

Closes #26

### CODE REVIEW

**Comment by abuzarmahmood** [comment]
## Solution Description

This PR implements a comprehensive solution for issue #26 by adding a new **LatentPoolCTRNN** model with the following capabilities:

### 1. Large Pool of Latent Neurons
- Configurable latent neuron pool (default: 256 neurons)
- Learns shared dynamics across multiple recording datasets
- Uses continuous-time RNN (CTRNN) architecture for temporal dynamics

### 2. Multi-Dataset Training
- Supports training on multiple datasets simultaneously
- Handles variable numbers of observed neurons per dataset
- Uses masking mechanism to handle padding for datasets with fewer neurons
- Encoder/decoder architecture adapts to different neuron counts

### 3. Stimulus Integration
- **Stimulus Type**: One-hot encoded input indicating which stimulus was presented
- **Stimulus Concentration**: Continuous input for stimulus intensity/concentration
- Both inputs are concatenated with spike counts before encoding

### 4. Smooth Firing Rate Estimation
- Takes binned spike counts as input
- Outputs smooth firing rate estimates
- Uses temporal smoothness constraint in loss function (smooth_MSELoss)
- ReLU activation ensures non-negative firing rates

### Implementation Details

**Model Architecture:**
1. **Encoder**: Projects [spike_counts + stimulus_type + stimulus_concentration] to latent space
2. **CTRNN**: Processes latent representations over time with continuous dynamics
3. **Decoder**: Projects latent space back to firing rate estimates

**Training Features:**
- Cross-validation support
- Early stopping with configurable patience
- Model checkpointing (saves best model based on validation loss)
- Handles multiple datasets in a single training loop
- Dropout regularization

**Example Usage:**
The `src/run_latent_pool_model.py` script demonstrates:
- Creating synthetic data for multiple datasets with different neuron counts
- Preparing data with stimulus information
- Training the model
- Evaluating performance with R² scores
- Generating training plots

### Testing
Run the example script to verify the implementation:
```bash
python src/run_latent_pool_model.py
```

This will:
- Generate 3 synthetic datasets with 30, 25, and 35 neurons respectively
- Train a LatentPoolCTRNN with 128 latent neurons
- Output training progress and validation metrics
- Save model checkpoints and training plots

### Files Modified/Added
- `src/model.py`: Added LatentPoolCTRNN class
- `src/train.py`: Added train_latent_pool_model function
- `src/run_latent_pool_model.py`: New example script (can be run independently)

All requirements from issue #26 have been addressed.
